src/internal/transactionenv/env.go

```diff
@@ -141,6 +141,12 @@ type AuthTransactionServer interface {
 	ModifyRoleBindingInTransaction(*TransactionContext, *auth.ModifyRoleBindingRequest) (*auth.ModifyRoleBindingResponse, error)
 	GetRoleBindingInTransaction(*TransactionContext, *auth.GetRoleBindingRequest) (*auth.GetRoleBindingResponse, error)
 
+	// Methods to add and remove pipelines from input and output repos. These do their own auth checks
+	// for specific permissions required to use a repo as a pipeline input/output.
+	AddPipelineReaderToRepoInTransaction(*TransactionContext, string, string) error
+	AddPipelineWriterToRepoInTransaction(*TransactionContext, string) error
+	RemovePipelineReaderFromRepoInTransaction(*TransactionContext, string, string) error
```

Comment:
These are very specific methods in the internal API that have their own auth. The alternative was to put the auth logic in PPS and expose an API that supports modifying all role bindings without auth, which feels like a bigger possible footgun.